gcp/ignore_error_predicate.go

```diff
@@ -23,12 +24,21 @@ func isIgnorableError(notFoundErrors []string) plugin.ErrorPredicate {
 // shouldIgnoreErrorPluginDefault:: Plugin level default function to ignore a set errors for hydrate functions based on "ignore_error_codes" config argument
 func shouldIgnoreErrorPluginDefault() plugin.ErrorPredicateWithContext {
 	return func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, err error) bool {
-		if !hasIgnoredErrorCodes(d.Connection) {
-			return false
-		}
-
 		gcpConfig := GetConfig(d.Connection)
+
 		if gerr, ok := err.(*googleapi.Error); ok {
+			for _, pattern := range gcpConfig.IgnoreErrorMessages {
+				re := regexp.MustCompile(pattern)
+				result := re.FindString(types.ToString(gerr.Message))
```

Comment:
Does this function support matching on full regular expressions? How is this different than the regex `Match` function?

Reply:
Yes, this function support matching on full regular expressions.

- **func (re *Regexp) FindString(s string) string**
FindString returns a string holding the text of the leftmost match in s of the regular expression. If there is no match, the return value is an empty string, but it will also be empty if the regular expression successfully matches an empty string. Use FindStringIndex or FindStringSubmatch if it is necessary to distinguish these cases.

- **func (re *Regexp) Match(b []byte) bool**
Match reports whether the byte slice b contains any match of the regular expression re.

IMO, it would be better to use `MatchString` function here. I can see in most of the place we use `MatchString` function to validate patterns. https://github.com/search?q=repo%3Aturbot%2Fsteampipe%20MatchString&type=code

I have replace the function `FindString` to `MatchString`.